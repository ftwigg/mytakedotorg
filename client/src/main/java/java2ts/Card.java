/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytake/mytake/graphs/contributors
 */
package java2ts;

import com.jsoniter.any.Any;

@jsweet.lang.Interface
public class Card {
	public String username;
	public String titleSlug;
	public String title;
	public Any blocks;
	public int[] previewBlocks;
}
