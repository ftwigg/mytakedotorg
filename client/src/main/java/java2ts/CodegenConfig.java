/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytake/mytake/graphs/contributors
 */
package java2ts;

import com.jsoniter.spi.TypeLiteral;
import com.jsoniter.static_codegen.StaticCodegenConfig;

@jsweet.lang.Erased
public class CodegenConfig implements StaticCodegenConfig {
	@Override
	public void setup() {
		// calls to JsoniterSpi
	}

	@SuppressWarnings("rawtypes")
	@Override
	public TypeLiteral[] whatToCodegen() {
		return new TypeLiteral[]{
				TypeLiteral.create(DraftRev.class),
				TypeLiteral.create(DraftPost.class),
				TypeLiteral.create(LoginCookie.class),
				TypeLiteral.create(PublishResult.class)
		};
	}
}
