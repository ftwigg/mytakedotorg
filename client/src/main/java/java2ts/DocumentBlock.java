/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytakedotorg/mytakedotorg/graphs/contributors
 */
package java2ts;

@jsweet.lang.Interface
public class DocumentBlock {
	public StringTypes.document kind;
	public String excerptId;
	public jsweet.util.tuple.Tuple2<Integer, Integer> highlightedRange;
	public jsweet.util.tuple.Tuple2<Integer, Integer> viewRange;
}
