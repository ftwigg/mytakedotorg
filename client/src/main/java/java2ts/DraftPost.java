/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytakedotorg/mytakedotorg/graphs/contributors
 */
package java2ts;

import com.jsoniter.any.Any;

@jsweet.lang.Interface
public class DraftPost implements Json {
	@jsweet.lang.Optional
	public DraftRev parentRev;
	public String title;
	public Any blocks;
}
