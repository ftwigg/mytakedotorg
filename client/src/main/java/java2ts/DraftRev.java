/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytake/mytake/graphs/contributors
 */
package java2ts;

@jsweet.lang.Interface
public class DraftRev implements Json {
	public int draftid;
	public int lastrevid;
}
