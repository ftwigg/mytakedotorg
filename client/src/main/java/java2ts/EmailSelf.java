/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytakedotorg/mytakedotorg/graphs/contributors
 */
package java2ts;

@jsweet.lang.Interface
public class EmailSelf implements Json {
	public String subject;
	public String body;
}
