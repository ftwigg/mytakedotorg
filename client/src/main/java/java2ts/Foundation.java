/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytake/mytake/graphs/contributors
 */
package java2ts;

import java.util.Date;
import java.util.List;
import java2ts.StringTypes.recorded;
import jsweet.util.tuple.Tuple2;

public interface Foundation {
	///////////////////////////////
	// Unused, but what I'd like //
	///////////////////////////////
	@jsweet.lang.Interface
	public class Fact {
		public String title;
		public String primaryDate; // yyyy-mm-dd
		public String primaryDateKind;
		public String kind;
	}

	@jsweet.lang.Interface
	public class FactLink {
		public Fact fact;
		public String hash;
	}

	@jsweet.lang.Interface
	public class FactContent {
		public Fact fact;
	}

	@jsweet.lang.Interface
	public class VideoFactContent extends FactContent {
		public String youtubeId;
		@jsweet.lang.Optional
		public List<Person> speakers;
		@jsweet.lang.Optional
		public List<CaptionWord> transcript;
		@jsweet.lang.Optional
		public List<SpeakerMap> speakerMap;
	}

	@jsweet.lang.Interface
	public class DocumentFactContent extends FactContent {
		public List<DocumentComponent> components;
	}

	@jsweet.lang.Interface
	public class DocumentComponent {
		public String component;
		public String innerHTML;
		public int offset;
	}

	//////////////////
	// What we have //
	//////////////////
	@jsweet.lang.Interface
	public class DocumentFact {
		public String title;
		public String filename;
		public Date primaryDate;
		// TODO: https://github.com/cincheo/jsweet/issues/412
		public String primaryDateKind;
	}

	@jsweet.lang.Interface
	public class VideoFact {
		public String id;
		public String title;
		public Date primaryDate;
		public recorded primaryDateKind;
	}

	@jsweet.lang.Interface
	public class Person {
		public String firstname;
		@jsweet.lang.Optional
		public String middlename;
		public String lastname;
	}

	@jsweet.lang.Interface
	public class SpeakerMap {
		public String speaker; // lastname of Person
		public Tuple2<Integer, Integer> range;
	}

	@jsweet.lang.Interface
	public class CaptionWord {
		public int idx;
		public String word;
		public double timestamp;
	}

	@jsweet.lang.Interface
	public class CaptionMeta {
		public List<Person> speakers;
		public List<SpeakerMap> speakerMap;
	}
}
