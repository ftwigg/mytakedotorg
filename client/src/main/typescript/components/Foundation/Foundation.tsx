import * as React from "react";
import * as ReactDOM from "react-dom";
import TimelineView from "../TimelineView";

interface FoundationProps {}

class Foundation extends React.Component<FoundationProps, {}> {
  constructor(props: FoundationProps) {
    super(props);
  }
  render() {
    return (
      <div className="foundation">
        <div className="foundation__inner-container">
          <h1 className="foundation__heading">Foundation of Facts</h1>
          <p className="timeline__instructions">
            Explore Facts in the timeline below.
          </p>
        </div>
        <TimelineView />
      </div>
    );
  }
}

export default Foundation;
