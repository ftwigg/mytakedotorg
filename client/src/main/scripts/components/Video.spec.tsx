import * as React from "react";
import * as renderer from "react-test-renderer";
import {} from "jest";
import Video from "./Video";
import { Foundation } from "../java2ts/Foundation";

jest.mock("react-youtube", () => ({
  default: "YouTube"
}));

jest.mock("./CaptionView", () => ({
  default: "CaptionView"
}));

const mockFn = jest.fn();

const videoFact: Foundation.VideoFactContent = {
  youtubeId: "ApTLB76Nmdg",
  speakers: [
    {
      firstname: "Martha",
      middlename: null,
      lastname: "Raddatz"
    }
  ],
  speakerMap: [
    {
      speaker: "Raddatz",
      range: [0, 7]
    }
  ],
  transcript: [
    {
      idx: 0,
      word: "Good ",
      timestamp: 0.75
    },
    {
      idx: 1,
      word: "evening ",
      timestamp: 0.96
    },
    {
      idx: 2,
      word: "I'm ",
      timestamp: 1.56
    },
    {
      idx: 3,
      word: "Martha ",
      timestamp: 1.92
    },
    {
      idx: 4,
      word: "Raddatz ",
      timestamp: 2.129
    },
    {
      idx: 5,
      word: "from ",
      timestamp: 2.52
    },
    {
      idx: 6,
      word: "ABC ",
      timestamp: 3.06
    },
    {
      idx: 7,
      word: "News. ",
      timestamp: 3.09
    }
  ],
  fact: {
    title: "Donald Trump - Hillary Clinton (2/3)",
    primaryDate: "2016-10-09",
    primaryDateKind: "recorded",
    kind: "video"
  }
};

test("Video", () => {
  const tree = renderer
    .create(
      <Video
        videoFact={videoFact}
        onSetClick={mockFn}
        className={"video__inner-container"}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Video with highlights", () => {
  const tree = renderer
    .create(
      <Video
        videoFact={videoFact}
        onSetClick={mockFn}
        className={"video__inner-container"}
        timeRange={[0, 3]} // Test range can't go outside of videoFact.transcript range
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
