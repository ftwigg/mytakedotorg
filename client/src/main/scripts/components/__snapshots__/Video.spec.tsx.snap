// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video 1`] = `
<div
  className="video__outer-container"
>
  <div
    className="video__inner-container"
  >
    <div
      className="video__container"
    >
      <div
        className="video__header"
      >
        <p
          className="video__date"
        >
          2016-10-09
        </p>
      </div>
      <YouTube
        className="video__video"
        onPause={[Function]}
        onReady={[Function]}
        onStateChange={[Function]}
        opts={
          Object {
            "height": "315",
            "playerVars": Object {
              "autoplay": 1,
              "cc_lang_pref": "en",
              "cc_load_policy": 1,
              "modestbranding": 1,
              "playsinline": 1,
              "rel": 0,
              "showinfo": 0,
              "start": 0,
            },
            "width": "560",
          }
        }
        videoId="ApTLB76Nmdg"
      />
    </div>
    <CaptionView
      captionIsHighlighted={false}
      eventHandlers={
        Object {
          "onClearPress": [Function],
          "onCursorPlace": [Function],
          "onFineTuneDown": [Function],
          "onFineTuneUp": [Function],
          "onHighlight": [Function],
        }
      }
      highlightedCharRange={
        Array [
          -1,
          -1,
        ]
      }
      timer={0}
      videoEnd={-1}
      videoFact={
        Object {
          "fact": Object {
            "kind": "video",
            "primaryDate": "2016-10-09",
            "primaryDateKind": "recorded",
            "title": "Donald Trump - Hillary Clinton (2/3)",
          },
          "speakerMap": Array [
            Object {
              "range": Array [
                0,
                7,
              ],
              "speaker": "Raddatz",
            },
          ],
          "speakers": Array [
            Object {
              "firstname": "Martha",
              "lastname": "Raddatz",
              "middlename": null,
            },
          ],
          "transcript": Array [
            Object {
              "idx": 0,
              "timestamp": 0.75,
              "word": "Good ",
            },
            Object {
              "idx": 1,
              "timestamp": 0.96,
              "word": "evening ",
            },
            Object {
              "idx": 2,
              "timestamp": 1.56,
              "word": "I'm ",
            },
            Object {
              "idx": 3,
              "timestamp": 1.92,
              "word": "Martha ",
            },
            Object {
              "idx": 4,
              "timestamp": 2.129,
              "word": "Raddatz ",
            },
            Object {
              "idx": 5,
              "timestamp": 2.52,
              "word": "from ",
            },
            Object {
              "idx": 6,
              "timestamp": 3.06,
              "word": "ABC ",
            },
            Object {
              "idx": 7,
              "timestamp": 3.09,
              "word": "News. ",
            },
          ],
          "youtubeId": "ApTLB76Nmdg",
        }
      }
      videoStart={0}
    />
  </div>
</div>
`;

exports[`Video with highlights 1`] = `
<div
  className="video__outer-container"
>
  <div
    className="video__inner-container"
  >
    <div
      className="video__container"
    >
      <div
        className="video__header"
      >
        <button
          className="video__button video__button--top"
          onClick={[Function]}
        >
          Give your Take on this
        </button>
        <p
          className="video__date"
        >
          2016-10-09
        </p>
      </div>
      <YouTube
        className="video__video"
        onPause={[Function]}
        onReady={[Function]}
        onStateChange={[Function]}
        opts={
          Object {
            "height": "315",
            "playerVars": Object {
              "autoplay": 1,
              "cc_lang_pref": "en",
              "cc_load_policy": 1,
              "end": 3,
              "modestbranding": 1,
              "playsinline": 1,
              "rel": 0,
              "showinfo": 0,
              "start": 0,
            },
            "width": "560",
          }
        }
        videoId="ApTLB76Nmdg"
      />
    </div>
    <CaptionView
      captionIsHighlighted={true}
      eventHandlers={
        Object {
          "onClearPress": [Function],
          "onCursorPlace": [Function],
          "onFineTuneDown": [Function],
          "onFineTuneUp": [Function],
          "onHighlight": [Function],
        }
      }
      highlightedCharRange={
        Array [
          0,
          37,
        ]
      }
      timer={0}
      videoEnd={3}
      videoFact={
        Object {
          "fact": Object {
            "kind": "video",
            "primaryDate": "2016-10-09",
            "primaryDateKind": "recorded",
            "title": "Donald Trump - Hillary Clinton (2/3)",
          },
          "speakerMap": Array [
            Object {
              "range": Array [
                0,
                7,
              ],
              "speaker": "Raddatz",
            },
          ],
          "speakers": Array [
            Object {
              "firstname": "Martha",
              "lastname": "Raddatz",
              "middlename": null,
            },
          ],
          "transcript": Array [
            Object {
              "idx": 0,
              "timestamp": 0.75,
              "word": "Good ",
            },
            Object {
              "idx": 1,
              "timestamp": 0.96,
              "word": "evening ",
            },
            Object {
              "idx": 2,
              "timestamp": 1.56,
              "word": "I'm ",
            },
            Object {
              "idx": 3,
              "timestamp": 1.92,
              "word": "Martha ",
            },
            Object {
              "idx": 4,
              "timestamp": 2.129,
              "word": "Raddatz ",
            },
            Object {
              "idx": 5,
              "timestamp": 2.52,
              "word": "from ",
            },
            Object {
              "idx": 6,
              "timestamp": 3.06,
              "word": "ABC ",
            },
            Object {
              "idx": 7,
              "timestamp": 3.09,
              "word": "News. ",
            },
          ],
          "youtubeId": "ApTLB76Nmdg",
        }
      }
      videoStart={0}
    />
  </div>
</div>
`;
